src/main/scala/org/hammerlab/guacamole/reads/Read.scala

```diff
@@ -85,22 +83,72 @@ object Read extends Logging {
+   * that field. The result is the intersection of the filters (i.e. reads must satisfy ALL filters).
    *
-   * @param mapped include only mapped reads
+   * @param overlapsLoci if not None, include only mapped reads that overlap the given loci
    * @param nonDuplicate include only reads that do not have the duplicate bit set
    * @param passedVendorQualityChecks include only reads that do not have the failedVendorQualityChecks bit set
    * @param isPaired include only reads are paired-end reads
+   * @param hasMdTag include only reads with md tags
    */
   case class InputFilters(
-    mapped: Boolean = false,
-    nonDuplicate: Boolean = false,
-    passedVendorQualityChecks: Boolean = false,
-    isPaired: Boolean = false,
-    hasMdTag: Boolean = false) {}
+    val overlapsLoci: Option[LociSet.Builder],
+    val nonDuplicate: Boolean,
+    val passedVendorQualityChecks: Boolean,
+    val isPaired: Boolean,
+    val hasMdTag: Boolean) {}
   object InputFilters {
     val empty = InputFilters()
+
+    /**
+     * See InputFilters for full documentation.
+     * @param mapped include only mapped reads. Convenience argument that is equivalent to specifying all sites in
+     *               overlapsLoci.
+     */
+    def apply(mapped: Boolean = false,
+              overlapsLoci: Option[LociSet.Builder] = None,
+              nonDuplicate: Boolean = false,
+              passedVendorQualityChecks: Boolean = false,
+              isPaired: Boolean = false,
+              hasMdTag: Boolean = false): InputFilters = {
+      new InputFilters(
+        overlapsLoci = if (overlapsLoci.isEmpty && mapped) Some(LociSet.newBuilder.putAllContigs) else overlapsLoci,
+        nonDuplicate = nonDuplicate,
+        passedVendorQualityChecks = passedVendorQualityChecks,
+        isPaired = isPaired,
+        hasMdTag = hasMdTag)
+    }
+
+    /**
+     * Apply filters to an RDD of reads.
+     *
+     * @param filters
+     * @param reads
+     * @param sequenceDictionary
+     * @return filtered RDD
+     */
+    def filterRDD(filters: InputFilters, reads: RDD[Read], sequenceDictionary: SequenceDictionary): RDD[Read] = {
+      /* Note that the InputFilter properties are public, and some loaders directly apply
+       * the filters as the reads are loaded, instead of filtering an existing RDD as we do here. If more filters
+       * are added, be sure to update those implementations.
+       * 
+       * This is implemented as a static function instead of a method in InputFilters because the overlapsLoci
+       * attribute cannot be serialized.
+       */
+      var result = reads
+      if (filters.overlapsLoci.nonEmpty) {
+        val loci = filters.overlapsLoci.get.result(contigLengths(sequenceDictionary))
+        val broadcastLoci = reads.sparkContext.broadcast(loci)
+        result = result.filter(
+          read => read.isMapped && read.asMappedRead.get.overlapsLociSet(broadcastLoci.value))
+      }
```

Comment:
Yeah, it's awkward because we can have plain `MappedRead` instances but also `PairedRead[MappedRead]`, so can't easily match on the type. The `asMappedRead` method deals with this issue.